### PR TITLE
new(pkg/driver): fixed some kernel version related issues.

### PR DESCRIPTION
**Pull request by FedeDP**
**What type of PR is this?**

/kind bug

**Any specific area of the project related to this PR?**

/area library
/area tests

**What this PR does / why we need it**:

Store full kernelversion in `driverkernel.FetchInfo()`, instead of already modifying it.
This way, all of the distro-specific logic can assume it receives the full `uname -v` output.
This fixes a bug with debian distro.

Also, properly `QueryEscape` driver version while building up the download url, because it contains a `+driver`.
This fixes drivers download.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

File: pkg/driver/distro/distro.go [hunk 1]
```diff
@@ -125,7 +126,7 @@ func getOSReleaseDistro(kr *kernelrelease.KernelRelease) (Distro, error) {
 }
 
 func toURL(repo, driverVer, fileName, arch string) string {
-	return fmt.Sprintf("%s/%s/%s/%s", repo, driverVer, arch, fileName)
+	return fmt.Sprintf("%s/%s/%s/%s", repo, url.QueryEscape(driverVer), arch, fileName)
```
Comment (FedeDP) [inline pkg/driver/distro/distro.go:129]: Also, properly escape the driverVersion that contains a `+driver`.

**Review by alacuku — COMMENTED** [review]
@FedeDP, thanks for the fixes. Left some comments.

File: pkg/driver/distro/debian.go [hunk 2]
```diff
@@ -68,10 +68,13 @@ func (d *debian) FixupKernel(kr kernelrelease.KernelRelease) kernelrelease.Kerne
 		archExtra = fmt.Sprintf("-%s%s", matches[1], matches[2])
 	}
 	if debianKernelVersionRegex.MatchString(kr.KernelVersion) {
+		newKV := debianKernelVersionRegex.FindStringSubmatch(kr.KernelVersion)[0]
```
[the remark this answers is not in the document]
Comment (alacuku) [reply pkg/driver/distro/debian.go:71]: You're right, no additional check is needed.

File: pkg/driver/distro/generic.go [hunk 3]
```diff
@@ -41,6 +43,10 @@ func (g *generic) String() string {
 
 //nolint:gocritic // the method shall not be able to modify kr
 func (g *generic) FixupKernel(kr kernelrelease.KernelRelease) kernelrelease.KernelRelease {
+	// Take eg: "#1 SMP PREEMPT_DYNAMIC Tue, 10 Oct 2023 21:10:21 +0000" and return "1".
+	kv := strings.Trim(kr.KernelVersion, "#")
```
[the remark this answers is not in the document]
Comment (FedeDP) [reply pkg/driver/distro/generic.go]: Done!

**Review by alacuku — APPROVED** [review]

**Review by leogr — APPROVED** [review]

**Pull request merged**